contracts/FragmentNFT.sol

```diff
@@ -187,7 +187,8 @@ contract FragmentNFT is IFragmentNFT, ERC721Upgradeable, ERC2771ContextExternalF
    */
   function tagCountAt(uint256 snapshotId) external view returns (bytes32[] memory tags_, uint256[] memory counts) {
     if (snapshotId >= _snapshots.length) revert BAD_SNAPSHOT_ID(snapshotId, _snapshots.length);
-    EnumerableMap.Bytes32ToUintMap storage tagCount = _snapshots[snapshotId].totalTagCount;
+    EnumerableMap.Bytes32ToUintMap storage tagCount = _snapshots[_findAccountSnapshotId(address(this), snapshotId)]
+      .totalTagCount;
     tags_ = tagCount.keys();
```

Comment:
Hi @rafael-ab , If there's no contribution, the snapshot will be the snapshot taken while minting the initial fragment ? Just to make sure, there will be one at least

Reply:
Yes, that's correct, there no will be empty values for previous snapshots because snapshots starts from 1

Reply:
LGTM!